_studio/shared/src/libmfx_allocator_vaapi.cpp

```diff
@@ -637,51 +637,59 @@ mfxDefaultAllocatorVAAPI::LockFrameHW(
     mfxFrameData*  ptr)
 {
     MFX_CHECK(pthis, MFX_ERR_INVALID_HANDLE);
-    mfxWideHWFrameAllocator *pSelf = (mfxWideHWFrameAllocator*)pthis;
+    MFX_CHECK(mid, MFX_ERR_INVALID_HANDLE);
+    MFX_CHECK_NULL_PTR1(ptr);
 
-    mfxStatus mfx_res = MFX_ERR_NONE;
-    VAStatus  va_res  = VA_STATUS_SUCCESS;
-    vaapiMemIdInt* vaapi_mids = (vaapiMemIdInt*)mid;
-    mfxU8* pBuffer = 0;
+    auto vaapi_mids = reinterpret_cast<vaapiMemIdInt*>(mid);
+    MFX_CHECK(vaapi_mids->m_surface, MFX_ERR_INVALID_HANDLE);
 
-    if (!vaapi_mids || !(vaapi_mids->m_surface)) return MFX_ERR_INVALID_HANDLE;
+    auto pSelf = reinterpret_cast<mfxWideHWFrameAllocator*>(pthis);
+
+    VAStatus va_res = VA_STATUS_SUCCESS;
 
     mfxU32 mfx_fourcc = ConvertVP8FourccToMfxFourcc(vaapi_mids->m_fourcc);
     if (MFX_FOURCC_P8 == mfx_fourcc)   // bitstream processing
     {
-        MFX_AUTO_LTRACE(MFX_TRACE_LEVEL_EXTCALL, "vaMapBuffer");
-        VACodedBufferSegment *coded_buffer_segment;
         if (vaapi_mids->m_fourcc == MFX_FOURCC_VP8_SEGMAP)
-            va_res =  vaMapBuffer(pSelf->pVADisplay, *(vaapi_mids->m_surface), (void **)(&pBuffer));
+        {
+            mfxU8* pBuffer = nullptr;
+            {
+                MFX_AUTO_LTRACE(MFX_TRACE_LEVEL_EXTCALL, "vaMapBuffer");
+                va_res = vaMapBuffer(pSelf->pVADisplay, *(vaapi_mids->m_surface), (void **)(&pBuffer));
+                MFX_CHECK(va_res == VA_STATUS_SUCCESS, VA_TO_MFX_STATUS(va_res));
+            }
+
+            ptr->Y = pBuffer;
```

Comment:
Do we allow double lock? Do we want to check it? (better in separate commit in the end of refactor to simplify bisect)

Reply:
I don't want to introduce the logic that wasn't initially in the design of this class. and previously this class didn't care about such things. We may think about it and add this functionality in separate PR.

Actually I don't know what will return ```vaMap```/```vaUnmap``` for already mapped/unmapped buffer